### PR TITLE
[SYNTAX] Create LinkTapArea

### PR DESCRIPTION
**Pull request by ChessyHsu**
[Jira ticket](https://cambly.atlassian.net/browse/WEB-6?atlOrigin=eyJpIjoiYWU0NTQ3YjllOTMyNDY3OWE1M2U5YjQxYWQ2NzUwYTEiLCJwIjoiaiJ9)

Create LinkTapArea by replacing div in TapArea with a tag. Now LinkTapArea behaves like a tag, it provides accessibility for users who rely on assistive technologies and shows url when you hover over it on the bottom left corner.

Changes:
- add LinkTapArea
- add LinkTapArea to storybook
- add unit tests for LinkTapArea
- run pnpm changeset

Screenshot 2024-10-31 at 3 18 50 PM

### CODE REVIEW

File: packages/syntax-core/src/LinkTapArea/LinkTapArea.test.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        target="_blank"
+        data-testid="link-tap-area-testid"
+        onClick={() => {
+          /* empty */
+        }}
+      />,
+    );
+    const linkTapArea = await screen.findByTestId("link-tap-area-testid");
+    expect(linkTapArea).toHaveAttribute("href", "https://www.google.com");
+  });
```
Comment (ChessyHsu) [inline packages/syntax-core/src/LinkTapArea/LinkTapArea.test.tsx:61]: These 3 are a tag related tests

File: packages/syntax-core/src/LinkTapArea/LinkTapArea.test.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+          /* empty */
+        }}
+        ref={ref}
+      />,
+    );
+    expect(ref.current instanceof HTMLAnchorElement).toBeTruthy();
+    expect(ref.current?.getAttribute("data-testid")).toStrictEqual(
+      "link-tap-area-testid",
+    );
+  });
```
Comment (ChessyHsu) [inline packages/syntax-core/src/LinkTapArea/LinkTapArea.test.tsx:81]: These 3 tests above are a tag related

File: packages/syntax-core/src/LinkTapArea/LinkTapArea.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+              Link:{" "}
+              <a href="https://www.cambly.com/learn" target="_blank">
+                Cambly
+              </a>
+            </Typography>
+          </Box>
+        </LinkTapArea>
+      </Box>
+    </>
+  ),
```
Comment (ChessyHsu) [inline packages/syntax-core/src/LinkTapArea/LinkTapArea.stories.tsx:175]: @somethiiing copied the same examples in TapArea over. Not sure if we want to demo any other attributes.

**Review by somethiiing — APPROVED** [review]
the next step is to make sure to create the 

NextLinkTapArea
ExpressLinkTapArea

 + adding the eslint rule for `no-restricted-imports`

**Pull request merged**